app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+} from 'react-native';
+import {intlShape} from 'react-intl';
+import * as Animatable from 'react-native-animatable';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+
+import {General} from 'mattermost-redux/constants';
+
+import FormattedText from 'app/components/formatted_text';
+import Loading from 'app/components/loading';
+import PostList from 'app/components/post_list';
+import PostListRetry from 'app/components/post_list_retry';
+import SafeAreaView from 'app/components/safe_area_view';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+Animatable.initializeRegistryWithDefinitions({
+    growOut: {
+        from: {
+            opacity: 1,
+            scale: 1
+        },
+        0.5: {
+            opacity: 1,
+            scale: 3
+        },
+        to: {
+            opacity: 0,
+            scale: 5
+        }
+    }
+});
+
+export default class Permalink extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getPostsAfter: PropTypes.func.isRequired,
+            getPostsBefore: PropTypes.func.isRequired,
+            getPostThread: PropTypes.func.isRequired,
+            getChannel: PropTypes.func.isRequired,
+            handleSelectChannel: PropTypes.func.isRequired,
+            handleTeamChange: PropTypes.func.isRequired,
+            joinChannel: PropTypes.func.isRequired,
+            loadThreadIfNecessary: PropTypes.func.isRequired,
+            markChannelAsRead: PropTypes.func.isRequired,
+            markChannelAsViewed: PropTypes.func.isRequired,
+            selectPost: PropTypes.func.isRequired,
+            setChannelDisplayName: PropTypes.func.isRequired,
+            setChannelLoading: PropTypes.func.isRequired
+        }).isRequired,
+        channelId: PropTypes.string,
+        channelName: PropTypes.string,
+        channelTeamId: PropTypes.string,
+        currentTeamId: PropTypes.string.isRequired,
+        currentUserId: PropTypes.string.isRequired,
+        focusedPostId: PropTypes.string.isRequired,
+        isPermalink: PropTypes.bool,
+        myMembers: PropTypes.object.isRequired,
+        navigator: PropTypes.object,
+        onClose: PropTypes.func,
+        onPermalinkPress: PropTypes.func,
+        onPress: PropTypes.func,
+        postIds: PropTypes.array,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        onPress: () => true,
+        postIds: []
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        const {postIds, channelName} = props;
+        let show = false;
+
+        if (postIds && postIds.length >= 10) {
+            show = true;
+        }
+
+        if (!show) {
+            this.loadPosts(props, context);
+        }
+
+        this.state = {
+            title: channelName,
+            show,
```

Comment:
This sounds a bit like this controls whether or not the whole screen is visible. Maybe something like `loading` would be more clear? 0/5 though